### PR TITLE
Fix for #1824 `s,http://rdlib.net,http://rdflib.net,g`

### PR DESCRIPTION
**Pull request by gjhiggins**
# Summary of changes

Fix for #1824 reported  typo of `http://rdlib.net/`,  change `http://rdlib.net` to `http://rdflib.net` in three files.

# Checklist
- [x] Checked that there aren't other open pull requests for the same change.
- [x] Considered granting [push permissions to the PR branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork),
  so maintainers can fix minor issues and keep your PR up to date.

### CODE REVIEW

**Review by edmondchuc — APPROVED** [review]

**Comment by edmondchuc** [comment]
Is this considered a breaking change?

**Comment by gjhiggins** [comment]
> Is this considered a breaking change?

Now that you mention it, probably yes, given that the skolemization of`timbl-card.n3`pre- and post-change yields graphs that are not `==`.

So, do I just close the PR and keep it around for when we're ready?

**Comment by aucampia** [comment]
> Given that neither domain is under the control of RDFLib, oughtn't we remake it further anyway? E.g. using `rdflib.dev` (if we bet on that being maintained during the lifetime of this project).

I'm in favor of this, we should not use domains we don't own so `rdflib.dev` will be better another option may be `rdflib.github.io`.

**Comment by gjhiggins** [comment]
> > Given that neither domain is under the control of RDFLib, oughtn't we remake it further anyway? E.g. using `rdflib.dev` (if we bet on that being maintained during the lifetime of this project).
> 
> I'm in favor of this, we should not use domains we don't own so `rdflib.dev` will be better another option may be `rdflib.github.io`.

Changed.

**Comment by aucampia** [comment]
This PR includes the following changes with runtime impact:

- Change the binding for `genid` that gets set under some circumstances: https://github.com/RDFLib/rdflib/blob/c906f64b915aee28a57aae74819884fd3cab74ab/rdflib/graph.py#L2051-L2053

  I would consider this a breaking change as under some strange circumstances this could result in user code doing something different than before, but technically it was wrong before as it should match the prefixed used by `BNode().skolemize()` to have the desired effect, and it won't - so the current value is a bugged value to me anyway. I'm also not sure setting this in this place makes a lot of sense, I would say this should either always be set, or never. The current behavior for setting this seems somewhat arbitrary.

  **My determination**: Treat it as a non breaking change because the situations in which it would effect end user behavior is extremely arbitrary. In future we should re-consider what we are doing here though as I think it may be best to not do that, and either make `genid` part of our default namespace prefix set or not.

- Change the user-agent comment used in some contexts:

  https://github.com/RDFLib/rdflib/blob/c906f64b915aee28a57aae74819884fd3cab74ab/rdflib/parser.py#L185-L187

  The user agent comment has no specific semantics associated with it in IETF RFC 7231 [[ref](https://datatracker.ietf.org/doc/html/rfc7231#section-5.5.3)] - however many systems do effectively assign some semantics to it.  The comment in our user-agent however seems to not really follow the semantics that is assigned to it [[ref](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent)].

  **My determination**: Treat it as a non breaking change as nothing mandates that semantics be assigned to the user-agent comment and in this case it is unclear why someone would assign semantics to it. Our user-agent comment anyway seems entirely arbitrary. In future I would recommend removing the user-agent comment entirely as it is unclear what benefit it serves.

- Change the default skolemization authority: https://github.com/RDFLib/rdflib/blob/8cf7494ad6d25e8491ad216c347872b32f81383e/rdflib/term.py#L484-L485

  I don't see this as breaking, as there is nothing anywhere that I can find that assigns any semantic meaning to the authority used for generating Skolem IRIs. I also see our current value as a bug as we are not in control of the authority we currently use, which seems to go against what is said in the RDF 1.1 abstract syntax spec.

  **My determination**: Treat it as a non breaking change.


I would like to know if anyone differs on any of these points: @RDFLib/core

**Comment by gjhiggins** [comment]
> I would like to know if anyone differs on any of these points: @RDFLib/core

I'm in agreement.

**Review by aucampia — APPROVED** [review]
LGTM, as discussed earlier I don't think we should treat this as a breaking change, but open to more input on the matter.

File: rdflib/graph.py [hunk 1]
```diff
@@ -2049,7 +2049,9 @@ def graph(self, identifier=None, base=None):
             from rdflib.term import rdflib_skolem_genid
 
             self.bind(
-                "genid", "http://rdflib.net" + rdflib_skolem_genid, override=False
+                "genid",
+                "https://rdflib.github.io" + rdflib_skolem_genid,
```
[the remark this answers is not in the document]
Comment (aucampia) [reply rdflib/graph.py]: Added.

**Comment by aucampia** [comment]
@gjhiggins I also added an explicit test for this behavior:

https://github.com/RDFLib/rdflib/blob/7c51027d19cff07578aa5ad8bcd0d567c87024bb/rdflib/graph.py#L2047-L2056

Test is here: https://github.com/gjhiggins/rdflib/blob/fix-for-issue1824/test/test_dataset/test_dataset.py#L239-L268

**Review by niklasl — APPROVED** [review]
Very sound reasoning; I also agree!

**Pull request merged**